src/commands/php-errors.php

```diff
@@ -274,4 +218,99 @@ protected function execute( InputInterface $input, OutputInterface $output ) {
         $site_info->render();
 
     }
+
+	/**
+	 * Getting of sftp connection data for bot collaborator
+	 * @param $api_helper
+	 * @param $pressable_site_id
+	 *
+	 * @return array
+	 */
+	private function get_sftp_connection_data( $api_helper, $pressable_site_id ) {
+
+		$ftp_data = $api_helper->call_pressable_api( "sites/{$pressable_site_id}/ftp", 'GET', array() );
+
+		$ftp_config = array();
+
+		if ( empty( $ftp_data->data ) ) {
+			$ftp_config['error'] = 'Failed to retrieve FTP users. Aborting!';
+			return $ftp_config;
+		}
+
+		foreach ( $ftp_data->data as $ftp_user ) {
+			if ( PRESSABLE_BOT_COLLABORATOR_EMAIL === $ftp_user->email ) { // We found the bot collaborator we created, grab the info.
+				$ftp_config['sftp_username'] = $ftp_user->username;
+				$ftp_config['sftp_hostname'] = $ftp_user->sftpDomain;
+
+				$password_reset = $api_helper->call_pressable_api( "sites/{$pressable_site_id}/ftp/password/{$ftp_user->username}", 'POST', array() );
+				if ( ! empty( $password_reset->data ) ) {
+					$ftp_config['sftp_password'] = $password_reset->data;
+				} else {
+					$ftp_config['error'] = 'Failed to retrieve password for temporary bot collaborator. Aborting!';
+				}
+				break;
+			}
+		}
+
+		return $ftp_config;
+	}
+
+	/**
+	 * Add bot collaborator to site
+	 * Using batch_create, so we can set sftp_access role
+	 * @param $api_helper
+	 * @param $pressable_site_id
+	 *
+	 * @return array
+	 */
+	private function add_bot_collaborator_to_site( $api_helper, $pressable_site_id ) {
+
+		$response = array();
+
+		$add_collaborator_request = $api_helper->call_pressable_api(
+			"collaborators/batch_create",
+			'POST',
+			array(
+				'siteIds' => array(
+					$pressable_site_id,
+				),
+				'email'   => PRESSABLE_BOT_COLLABORATOR_EMAIL,
+				'roles'   => 'sftp_access',
+			)
+		);
+
+		if ( ! is_null( $add_collaborator_request->errors ) ) {
+			$response['error'] = 'Error creating temporary bot collaborator. Aborting!';
+			return $response;
+		}
+		// Get new collaborator.
```

Comment:
So, the reason we did this was so that later on we could delete the user, which we are no longer doing. Also, the loop is there because the API was slow to respond. The second bullet point in [this PR description](https://github.com/Automattic/team51-cli/pull/82#issue-1198678522) explains it. 

Just a thought here.... since we no longer need the user id, but we still want to confirm they've been added AND get their SFTP credentials, we could instead call your new `get_sftp_connection_data()` function instead of calling the pressable API to check on the user id on L270, and then return those SFTP credentials from this ( `add_bot_collaborator_to_site()` ) function. It would still be in the loop in case the delay still happens, and would cut down on an extra API call. What do you think?

Reply:
Nice catch! I'll make the changes in the next commit!

Reply:
@nickpagz this has been updated, can you re-check? I've divided the `add_bot_collaborator_to_site()` and getting the SFTP data after adding into separate functions so we respect [SRP](https://en.wikipedia.org/wiki/Single-responsibility_principle)

Reply:
Nice, works great in my testing. TIL about SRP, thanks!!